lib/logger.py

```diff
@@ -0,0 +1,80 @@
+import logging
+import logging.config
+
+class SeleniumFilter(logging.Filter):
+    def filter(self, record):
+        selenium_log = 'selenium.webdriver.remote.remote_connection'
+        return selenium_log not in record.getMessage()
+
+# much of this taken from http://victorlin.me/posts/2012/08/26/good-logging-practice-in-python
+# reference -- https://docs.python.org/2/library/logging.config.html
+logging.config.dictConfig({
+    "version": 1,
+    "disable_existing_loggers": False,
+    "formatters": {
+        "simple": {
+            "format": "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+        },
+        "console": {
+            "format": "%(message)s"
+        }
+    },
+
+    "handlers": {
+        "console": {
+            "class": "logging.StreamHandler",
+            "level": "INFO",
+            "formatter": "console",
+            "stream": "ext://sys.stdout"
+        },
+
+        "debug_file_handler": {
+            "class": "logging.handlers.RotatingFileHandler",
+            "level": "DEBUG",
+            "formatter": "simple",
+            "filename": "debug.log",
+            "maxBytes": 400000,
+            "backupCount": 20,
+            "encoding": "utf8"
+        },
+
+        "info_file_handler": {
+            "class": "logging.handlers.RotatingFileHandler",
+            "level": "INFO",
+            "formatter": "simple",
+            "filename": "info.log",
+            "maxBytes": 100000,
+            "backupCount": 20,
+            "encoding": "utf8"
+        },
+
+        "error_file_handler": {
+            "class": "logging.handlers.RotatingFileHandler",
+            "level": "ERROR",
+            "formatter": "simple",
+            "filename": "errors.log",
+            "maxBytes": 50000,
+            "backupCount": 20,
+            "encoding": "utf8"
+        }
+    },
```

Comment:
These file handlers are for debug, info, and error messages. They use the more verbose `"simple"` formatter, and write out to a file.

The files are rotated automatically once they grow to a certain size (400KB, 100KB, and 50KB, respectively). If I remember correctly, it'll copy the `error.log` file over to `error{DATE}.log`, and then start over on a new `error.log` file.

It'll keep 20 of them at a time, and then start deleting the oldest ones.